src/vendor/github.com/xelabs/go-mysqlstack/sqlparser/depends/sqltypes/arithmetic.go

```diff
@@ -255,54 +268,71 @@ func divNumeric(v1, v2 numeric) (numeric, error) {
 func prioritize(v1, v2 numeric) (altv1, altv2 numeric) {
 	switch v1.typ {
 	case Int64:
-		if v2.typ == Uint64 || v2.typ == Float64 {
+		if v2.typ == Uint64 || v2.typ == Float64 || v2.typ == Decimal {
 			return v2, v1
 		}
 	case Uint64:
+		if v2.typ == Float64 || v2.typ == Decimal {
+			return v2, v1
+		}
+	case Decimal:
 		if v2.typ == Float64 {
 			return v2, v1
 		}
 	}
 	return v1, v2
 }
 
-func intPlusInt(v1, v2 int64) numeric {
-	result := v1 + v2
-	if v1 > 0 && v2 > 0 && result < 0 {
-		goto overflow
+func intPlusInt(v1, v2 int64) (numeric, error) {
+	if (v1 > 0 && v2 > math.MaxInt64-v1) || (v1 < 0 && v2 < math.MinInt64-v1) {
+		return numeric{}, fmt.Errorf("BIGINT.value.is.out.of.range.in: %v, %v", v1, v2)
 	}
-	if v1 < 0 && v2 < 0 && result > 0 {
-		goto overflow
-	}
-	return numeric{typ: Int64, ival: result}
 
-overflow:
-	return numeric{typ: Float64, fval: float64(v1) + float64(v2)}
+	return numeric{typ: Int64, ival: v1 + v2}, nil
 }
 
 func uintPlusInt(v1 uint64, v2 int64) (numeric, error) {
 	if v2 < 0 {
-		return numeric{}, fmt.Errorf("cannot add a negative number to an unsigned integer: %d, %d", v1, v2)
+		if uint64(-v2) > v1 {
+			return numeric{}, fmt.Errorf("BIGINT.UNSIGNED.value.is.out.of.range.in: %v, %v", v1, v2)
```

Comment:
1  + -2 , -2 is not BIGINT?

Reply:
unsigned plus signed, the result is unsigned

Reply:
> 1 + -2 , -2 is not BIGINT?

unsigned plus signed, the result is unsigned